Gems/LyShine/Code/Editor/EditorWindow.cpp

```diff
@@ -2195,7 +2195,10 @@ void EditorWindow::closeEvent(QCloseEvent* closeEvent)
     // Save the current window state
     SaveEditorWindowSettings();
 
-    QMainWindow::closeEvent(closeEvent);
+    // Instead of closing this window, replicate the action of unchecking UI Editor from the Editor toolbar by
+    // hiding the parent view pane instead
+    nativeParentWidget()->hide();
+    closeEvent->ignore();
```

Comment:
I'm not entirely sure if we want to isolate this fix for just linux with an #ifdef or not

Reply:
That's what I was going to suggest, especially if this is a temporary fix while the underlying issue is being investigated.

Reply:
Isn't it better to hide this window rather than close? it would make it seem like it reopens faster next time you use it since you don't need to recreate all the elements.

Reply:
In general, I think the concern is with performance. With the amount of tools we have, keeping them around when the user closes them could take some memory away, which is a waste if they don't intend to re-open them. Unless the UI Editor does something special that warrants having to get an exception, I'm not sure why it should be an outlier.
Even if there was a reason for the UI Editor not to close but be hidden, at that point I think it should be a viewpane option and not an ad-hoc fix.

This is a temporary patch though, so I guess the above would be a separate conversation entirely :)

Reply:
Just to re-emphasize the fact that hiding the window instead of closing is what happens now if you close the window through unchecking the UI editor in the menu bar. The close event only happens when you click the 'x' on the upper right hand corner. If anything, this patch makes the behavior consistent. The deeper issue needs to be investigated as to why on Linux destroying a re-creating this particular window is problematic